File: Banking-Kata/src/main/java/banking/service/CustomerService.java
package banking.service;

import banking.email.EmailService;
import banking.model.User;
import banking.persistence.Repository;

import java.util.UUID;

public class CustomerService {

  private Repository<User> userRepository;
  private EmailService emailService;

  public CustomerService(Repository<User> userRepository, EmailService emailService) {
    this.userRepository = userRepository;
    this.emailService = emailService;
  }

  public void updateEmailAddress(UUID userId, String emailAddress) {
    User user = userRepository.findOne(userId).orElseThrow(() -> new RuntimeException("Invalid userId"));
    User updatedUser = user.cloneWithEmailAddress(emailAddress);
    userRepository.save(updatedUser);
  }

  public UUID sendFraudAlert(UUID userId) {
    // TODO implement this method!
    return null;
  }

}

File: 07-Banking-Kata/csharp-dotnetcore2/BankingKataAPI/Services/CustomerService.cs
using System;
using BankingKataAPI.Models;
using BankingKataAPI.Persistence;

namespace BankingKataAPI.Services
{
    public class CustomerService
    {
        private readonly IRepository<User> _userRepository;
        private readonly IEmailService _emailService;

        public CustomerService(IRepository<User> userRepository, IEmailService emailService) {
            _userRepository = userRepository;
            _emailService = emailService;
        }

        public void UpdateEmailAddress(Guid userId, string emailAddress)
        {
            var user = _userRepository.FindOne(userId) ?? throw new Exception("Invalid userId");
            var updatedUser = user.cloneWithEmailAddress(emailAddress);
            _userRepository.Save(updatedUser);
        }

        public Guid SendFraudAlert(Guid userId) {
            // TODO implement this method!
            return Guid.Empty;
        }
    }
}
